Makefile

```diff
@@ -104,12 +80,11 @@ test_images:
 
 .PHONY: tidy
 tidy:
-	go mod tidy
-	cd v2 && go mod tidy
+	go mod tidy -compat=1.17
```

Comment:
Pin the Go version to avoid the following errors.

```shell
$ go mod tidy
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/config loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/codelocation loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/failer loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/remote loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/suite loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/testingtproxy loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/writer loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/reporters loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/reporters/stenographer loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/reporters/stenographer/support/go-colorable loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/types loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/remote imports
        github.com/onsi/ginkgo/internal/spec_iterator loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/suite imports
        github.com/onsi/ginkgo/internal/containernode loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/suite imports
        github.com/onsi/ginkgo/internal/leafnodes loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/suite imports
        github.com/onsi/ginkgo/internal/spec loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/internal/suite imports
        github.com/onsi/ginkgo/internal/specrunner loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4
github.com/kubeflow/mpi-operator/v2/test/e2e tested by
        github.com/kubeflow/mpi-operator/v2/test/e2e.test imports
        github.com/onsi/ginkgo imports
        github.com/onsi/ginkgo/reporters/stenographer/support/go-colorable imports
        github.com/onsi/ginkgo/reporters/stenographer/support/go-isatty loaded from github.com/onsi/ginkgo@v1.11.0,
        but go 1.16 would select v1.16.4

To upgrade to the versions selected by go 1.16:
        go mod tidy -go=1.16 && go mod tidy -go=1.17
If reproducibility with go 1.16 is not needed:
        go mod tidy -compat=1.17
For other options, see:
        https://golang.org/doc/modules/pruning
```